### DISCUSSION TITLE
Alpine.effect() does not trigger when combined with Alpine.store()

### DISCUSSION THREAD

**Discussion by boustanihani** [alpinejs/alpine#4484] [7. Bugs (archived)]
I assume the `Alpine.store()` is reactive by nature.
I just need a function to get triggered when an array of images gets empty, the array is already inside the `Alpine.store`.
Shouldn't the following code work (without the need of using `Alpine.reactive()` ?
Everything is working great, except the code inside `Alpine.effect()` =>

```
Alpine.store('images') = [];

Alpine.effect(() => {
    if (Alpine.store('images').length == 0) { doSomething() }
})
```

**Comment by ekwoka** [comment]
Strange I didn't see this, but this should show as an parsing error for invalid left hand assignment.

You can't assign something to the result of a function call.

**Reply by boustanihani** [reply]
You're right, my mistake.
